mm2src/adex_cli/src/scenarios/mm2_proc_mng.rs

```diff
@@ -100,15 +100,20 @@ pub fn start_process(mm2_cfg_file: &Option<String>, coins_file: &Option<String>,
     }
 
     let Ok(mm2_binary) = get_mm2_binary_path() else { return; };
+    if !mm2_binary.exists() {
+        error!("Failed to start mm2, no file: {mm2_binary:?}");
+        return;
+    }
     start_process_impl(mm2_binary);
 }
 
 #[cfg(all(unix, not(target_os = "macos")))]
 pub fn start_process_impl(mm2_binary: PathBuf) {
     let mut command = Command::new(&mm2_binary);
-    let program = mm2_binary
-        .file_name()
-        .map_or("Undefined", |name: &OsStr| name.to_str().unwrap_or("Undefined"));
+    let Some(program) = mm2_binary.file_name() else {
```

Comment:
I guess `else` branch of `let-else` statement here will never work ... bcz `mm2_binary.file_name()` will always return `Some("mm2")` or `Some("mm2.exe")` ... depends on platform, whether the file actually exists or not.

Reply:
I know it,  that is just paranoic mode

Reply:
Please consider using `expect` or `panic` under `unwrap_or_else`